specs/consensus/overview.md

```diff
@@ -437,18 +437,134 @@ height of consensus `h_p`.
+the fact (e.g., a late joiner cannot check validity of a value that processes have 
+decided some time ago). So this use is not encouraged, and we ignore it in the 
+remainder. However, for some applications this use case might still be beneficial: in
+this case it is important to understand that one needs to make an argument (which
+also needs to involve
+the environment) why it
+can be ensured that if `v` is a value returned
+by the  `getValue()` function of a correct process in the current height, then
+`valid(v)` evaluates to `true`.
+
+**Remark.** Point 1 above also forbids that the current consensus round influences
+validity. For instance, one may say to get out of the liveness issue from the previous
+remark, to just find all blocks valid that have been proposed after, say, round 10. 
+Similarly, one might consider more stricter validity requirements for the proposal in
+round 0 (the lucky path), while in unlucky situations one might want to simplify 
+reaching a decision by weakening validity. In general this leads to complexity down
+the road when someone reads the decisions much later, and needs to understand the 
+different semantics of different blocks based on the decision round. We strongly
+suggest to not use round numbers in validation of values for this reason.
+
+#### Backward compatibility of `valid(v)`
+
+**Requirement 1 (Fixing bugs).**
+There might be a bug in `valid`. Then it might be possible that due to a bug,
+`valid` returns `false` 
+for a value `v` proposed by a correct process, and we are stuck at a given height. 
+A way to get out of the 
+situation is to produce a new implementation `valid'` that returns `true` for `v`. So
+to be prepared for such a scenario we need to allow a change in the function.
+
+**Requirement 2 (Future use).**
+So if we allow changes to `valid`, we need to understand all uses of this function: Some
+synchronization protocols may use `valid` for consistency checks, for instance, if a node
+fell behind, it might need to learn several past decisions. In doing so, it typically also
+uses (the current version) of `valid` to check the decided values before accepting them.
+In this scenario, a value decided in the past (potentially using a now old version of 
+`valid`) should be deemed valid with the current version of the function. 
+
+These two requirements lead us to the following requirement on the implementations:
+we consider a sequence of `valid_i(v)` implementations, with increasing indexes `i`, 
+so that to represent multiple _backwards compatible_ implementations of the validity checks.
+Formally we require that
+
+`valid_i(v) == true` implies `valid_j(v) == true `, for `j > i` and every value `v`
+
+The logical implications allows to be more permissive in later versions (as required to
+fix the bug), while maintaining the need that newer versions allow to check validity
+of previously decided values.
+
+#### Backward compatibility of `valid(v, chain, height)`
```

Comment:
I see this section as an example of the requirement of the previous section.

Although I agree that it is only a legit example if `v` somehow included a height or `h_p` is also known by the `valid(v)` implementation. Not sure if we need this. Maybe a pointer to issue 510 is enough here.

Reply:
I removed the section and put the discussion in the linked issue